system/jlib/jthread.cpp

```diff
@@ -1299,8 +1334,42 @@ class CThreadPool: public CThreadPoolBase, implements IThreadPool, public CInter
         }
         return false;
     }
-};
+    void setPoolSize(unsigned newPoolSize)
+    {
+        CriticalBlock block(crit);
+        if (newPoolSize > defaultmax)
+        {
+            availsem.signal(newPoolSize - defaultmax);
```

Comment:
What if there are already 'burst' threads running?
Simply increasing the # of slots in availsem, will immediately allow new blocked threads to start
but if burst threads are already running, they should be considered as already occupying a slot.

The upshot of not taking into account already running threads when increasing the limit, is that increaing the pool size can immediately mean more new threads launch.

To test could:
1) setting limit to e.g. 2
2) starting 4 (with small pool delay), i.e. 2 are 'bursting' threads
3) setPoolSize(4)
4) start 2 more pool threads
5) After v. short delay, check how many running. We want there to be 4 thread running, but there will be 6.